doc/man/gnunet-delete.1
.TH GNUNET-DELETE "1" "05 Apr 2003" "GNUnet"
.SH NAME
gnunet\-delete \- a command line interface for deleting indexed files from GNUnet
.SH SYNOPSIS
.B gnunet\-delete
[\fIOPTIONS\fR] \-f FILENAME
.SH DESCRIPTION
.PP
gnunet\-delete is used for deleting indexed files from GNUnet.
.TP
\fB\-h\fR, \fB\-\-help\fR
print help page
.TP
\fB\-c \fIFILENAME\fR, \fB\-\-config=FILENAME\fR
use config file (defaults: ~/.gnunet/gnunet.conf)
.TP
\fB\-v\fR, \fB\-\-version\fR
print the version number
.TP
\fB\-V\fR, \fB\-\-verbose\fR
be berbose; prints progress information and the hash-pair, CRC and filesize that can be used to download the file
.TP
\fB\-f \fIFILENAME\fR, \fB\-\-file=FILENAME\fR
delete FILENAME (obligatory)
.TP
\fB\-L \fILOGLEVEL\fR, \fB\-\-loglevel=LOGLEVEL\fR
change the loglevel. Possible values for LOGLEVEL are NOTHING, FATAL, ERROR, FAILURE, WARNING, MESSAGE, INFO, DEBUG, CRON and EVERYTHING.
.SH NOTES
The option \-f is used to specify the name of the file that should be deleted.
.TP
You can only delete files that you indexed (or inserted?? FIXME!) and that you still have available locally in full.  You should use gnunet-delete on files that you indexed (not inserted) and that you are going to delete locally.
.TP
.SH FILES
.TP
~/.gnunet/gnunet.conf
GNUnet configuration file
.SH "REPORTING BUGS"
Report bugs by using mantis <https://gnunet.org/mantis/> or by sending electronic mail to <gnunet-developers@gnu.org>
.SH "SEE ALSO"
\fBgnunet\-gtk\fP(1), \fBgnunet\-insert\fP(1), \fBgnunet\-gtk\fP(1), \fBgnunet\-search\fP(1), \fBgnunet\-download\fP(1), \fBgnunet.conf\fP(5), \fBgnunetd\fP(1)


ChangeLog
Sun Dec 31 23:56:31 MST 2006
	ncurses may need "-lm" in order to link.
	Releasing GNUnet 0.7.1a.

Sat Dec 30 00:21:49 MST 2006
	DHT routing now seems to work (not integrated with file-sharing,
	only for gnunet-dht-query).

Fri Dec 29 09:38:53 MST 2006
	Added UPnP support to GNUnet (IPv4 only, #843).

Thu Dec 28 21:22:10 MST 2006
	Hostname resolution with gnunet-peer-info implemented (#1076).
	Proper linking of gnunet-pseudonym (#1161).
	Drop group permissions when changing user (#1162).
	gnunet-download support for directory files implemented (#1013).

Thu Dec 28 20:44:45 MST 2006
	Hopefully fixing build problems on certain systems with
	unusual installation of iconv.

Tue Dec 26 20:54:03 MST 2006
	Added additional gnunet-gtk options to client configuration
	specification file.

Sun Dec 24 18:06:04 MST 2006
	Limit memory utilization by select write queue.	
	Improved select write buffering (fewer malloc/free
	operations and less copying).

Sat Dec 23 22:12:17 MST 2006
	Updated various man pages and some configuration documentation.
	Fixed TCP transport (did not work with NAT if port was set to 0).
	Fixed verbose option for gnunet-transport-check.
	Updated hostlist scripts in contrib/ to reflect new protocol IDs.
	Allow aborting of gnunet-transport-check with CTRL-C.
	Allow testing of bi-directional transports behind NAT with
	gnunet-transport-check.
	
Fri Dec 22 19:21:25 MST 2006
	Added option to set maximum number of file descriptors
	(convenience for OS X users where the default is too low).
	Releasing GNUnet 0.7.1.

Thu Dec 21 20:03:07 MST 2006
	Fixed gnunet-insert "-e" option.  Fixed various
	crashes in client code.  Improved OS X portability.
	
Tue Dec 19 19:03:48 MST 2006
	Allowing GNUnet (without gnunet-setup) to be build
	on systems without guile.

Sun Dec 17 16:54:33 MST 2006
	Various improvements for OS X portability.
	CPU load detection fix for Solaris.

Sat Dec 16 11:42:13 MST 2006
	Bugfixes in collection library (gnunet-pseudonym).
	
Mon Dec 11 21:11:54 MST 2006
	Implemented ncurses/dialog based gnunet-setup wizard.
	Releasing GNUnet 0.7.1pre2.

Sun Dec 10 00:36:11 MST 2006
	Fixed load calculation code.  More aggressive utilization
	of available resources.

Mon Dec  4 13:24:12 MST 2006
	Releasing GNUnet 0.7.1pre1.

Wed Nov 29 22:20:26 MST 2006
	Fixed various bugs with (de)serialization of FSUI state
	and closing searches with associated downloads.  Extended
	FSUI test suite.

Mon Nov 27 08:51:46 MST 2006
	Added conversion of ECRS error messages to FSUI 
	error event messages.

Sun Nov 19 00:20:14 MST 2006
	Improved error handling.  gnunetd now can communicate
	text error messages to clients (not just error codes).
	
Wed Nov 15 23:17:36 MST 2006
	Fixed expiration time setting.  Improved bias of
	migration towards valuable content.
	Releasing GNUnet 0.7.1pre0.

Wed Nov  1 20:58:44 MST 2006
	Fixed MySQL database size underreporting.

Wed Nov  1 13:09:53 MST 2006
	Fixed some problems with index creation in sqlite
	datastore (discovered with new sqlite benchmarking
	code).  Performance should improve significantly
	(observed was a factor of 4-10 depending on 
	database size and operation).

Thu Oct 19 23:44:24 MDT 2006
	Completed huge update to FSUI API (not fully debugged).
	Major changes include:
	* clients can associate pointer with each action
	* abort methods can cancel action (but do not stop everything)
	* upload and unindex operations can now be suspended and resumed
	* cleaned up naming conventions
	
Thu Sep 14 23:44:17 MDT 2006
	Breaking UDP transport protocol compatibility -- some
	fields have to be switched around to make it work with
	the new select code.  Expect to see some warnings when
	interacting with 0.7.0 peers.

Tue Sep  5 21:28:25 PDT 2006
	Switched ncurses interface of gnunet-setup to use
	external dialog library (new dependency!).

Fri Aug 18 00:01:37 PDT 2006
	Swiched http bootstrap mechanism to using libcurl
	(new dependency!).

Sat May 20 08:37:02 PDT 2006
	Releasing GNUnet 0.7.0e.

Sun May 14 02:51:15 PDT 2006
	Major gnunet-setup GTK code cleanup (now uses libglade).

Sat May 13 19:35:49 PDT 2006
	Made headers more C++ friendly.     

Wed May 10 22:11:17 PDT 2006
	Fixing bug that could result in hanging the UDP transport;
	found by Luigi Auriemma (thanks!).

Sat May  6 00:17:59 PDT 2006
	Releasing GNUnet 0.7.0d.

Fri May  5 01:33:42 PDT 2006
	Fixed bug with priorities of IBlocks of uploaded content
	(priority was left at zero, making those blocks be 
	possibly discarded rather quickly once the datastore filled
	up).

Tue May  2 00:01:25 PDT 2006
	Fixed bug in gap where the tracking of query origins for
	optimizing routing paths lacked a simple assignment of the
	query origin (found by code inspection).  As a result,
	routing performance should improve further.  Also fixed
	small memory leak in gap and reduced memory consumption by
	fixing Mantis #1058.

Sat Apr 22 13:50:39 PDT 2006
	Fixed bug in util/cron.c where stopCron() would wait for an 
	unnecessary sleep to complete.

Sat Mar 18 12:54:11 PST 2006
	Releasing GNUnet 0.7.0c.

Thu Mar 16 22:14:01 PST 2006
	Fixing various routing problems (resulting in better utilization
	of bandwidth by routing more content and less queries).
	
Wed Mar 15 00:00:14 PST 2006
	Fixing various content migration bugs (one that disabled it,
	one that made it unlikely that migration was even tried, and
	then various crashes in formerly dead code).

Mon Jan 23 19:04:17 PST 2006
	Added Swedish translation.

Sun Jan  1 21:35:59 PST 2006
	Reduced amount of hashing done to be O(n) and not O(n^2) for
	n local search results (for example, for 100 results, this can
	make the difference between hashing 200 MB and hashing 20 MB).

Sat Dec 31 17:02:37 PST 2005
	Added support for using -k multiple times in gnunet-pseudonym.

Wen Dec 28 00:22:54 CET 2005
	Fixed deadlock in http transport.

Fri Dec 23 17:17:18 PST 2005
	Releasing GNUnet 0.7.0b.
	
Sat Nov 19 16:00:16 PST 2005
	Releasing GNUnet 0.7.0a.

Fri Oct  7 15:28:02 PDT 2005

src/applications/fs/tools/gnunet-delete.c
/*
     This file is part of GNUnet.
     (C) 2003 Christian Grothoff (and other contributing authors)

     GNUnet is free software; you can redistribute it and/or modify
     it under the terms of the GNU General Public License as published
     by the Free Software Foundation; either version 2, or (at your
     option) any later version.

     GNUnet is distributed in the hope that it will be useful, but
     WITHOUT ANY WARRANTY; without even the implied warranty of
     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
     General Public License for more details.

     You should have received a copy of the GNU General Public License
     along with GNUnet; see the file COPYING.  If not, write to the
     Free Software Foundation, Inc., 59 Temple Place - Suite 330,
     Boston, MA 02111-1307, USA.
*/

/**
 * Tool to delete files that were indexed with gnunet-insert.
 *
 * @author Christian Grothoff
 * @file applications/afs/tools/gnunet-delete.c
 */

#include "gnunet_afs_esed2.h"
#include "platform.h"

/**
 * Print progess message.
 */
static void printstatus(ProgressStats * stats,
			void * verboselevel) {
  if (*(int*)verboselevel == YES) {
    printf(_("%8u of %8u bytes deleted."),
	   (unsigned int) stats->progress,
	   (unsigned int) stats->filesize);
    printf("\r");
  }
}

/**
 * Prints the usage information for this command if the user errs.
 * Aborts the program.
 */
static void printhelp() {
  static Help help[] = {
    HELP_CONFIG,
    { 'f', "file", "NAME",
      gettext_noop("specify the file to delete from GNUnet (obligatory, file must exist)") } ,
    HELP_HELP,
    HELP_HOSTNAME,
    HELP_LOGLEVEL,
    HELP_VERSION,
    HELP_VERBOSE,
    HELP_END,
  };
  formatHelp("gnunet-delete [OPTIONS] -f FILENAME",
	     _("Remove file from GNUnet.  The specified file is not removed\n"
	       "from the filesystem but just from the local GNUnet datastore."),
	     help);
}

static int parseOptions(int argc,
			char ** argv) {
  int c;

  FREENONNULL(setConfigurationString("GNUNET-INSERT",
				     "INDEX-CONTENT",
				     "YES"));
  while (1) {
    int option_index=0;
    static struct GNoption long_options[] = {
      LONG_DEFAULT_OPTIONS,
      { "file",          1, 0, 'f' },
      { "verbose",       0, 0, 'V' },
      { 0,0,0,0 }
    };
    c = GNgetopt_long(argc,
		      argv,
		      "vhdc:L:H:Vf:",
		      long_options,
		      &option_index);
    if (c == -1)
      break;  /* No more flags to process */
    if (YES == parseDefaultOptions(c, GNoptarg))
      continue;
    switch(c) {
    case 'V':
      FREENONNULL(setConfigurationString("GNUNET-INSERT",
					 "VERBOSE",
					 "YES"));
      break;
    case 'f':
      FREENONNULL(setConfigurationString("GNUNET-DELETE",
					 "FILENAME",
					 GNoptarg));
      break;
    case 'v':
      printf("GNUnet v%s, gnunet-delete v%s\n",
	     VERSION,
	     AFS_VERSION);
      return SYSERR;
    case 'h':
      printhelp();
      return SYSERR;
    default:
      GE_LOG(ectx, GE_ERROR | GE_IMMEDIATE | GE_USER,
	  _("Use --help to get a list of options.\n"));
      return SYSERR;
    } /* end of parsing commandline */
  } /* while (1) */
  return OK;
}


/**
 * The main function to delete files from GNUnet.
 * @param argc number of arguments from the command line
 * @param argv command line arguments
 * @return return 0 for ok, -1 on error
 */
int main(int argc, char ** argv) {
  int beVerbose;
  GNUNET_TCP_SOCKET * sock;
  int ok;
  char * filename;

  if (SYSERR == initUtil(argc, argv, &parseOptions))
    return 0;
  beVerbose = testConfigurationString("GNUNET-INSERT",
				      "VERBOSE",
				      "YES");

  filename = getFileName("GNUNET-DELETE",
			 "FILENAME",
			 _("You must specify a filename (option -f)\n"));
  sock = getClientSocket();
  if (sock == NULL)
    errexit(_("Could not connect to gnunetd.\n"));
  ok = deleteFile(sock,
		  filename,
		  &printstatus,
		  &beVerbose);
  if (ok != OK) {
    GE_LOG(ectx, GE_DEBUG | GE_REQUEST | GE_USER,
	"Error deleting file `%s'.\n",
	filename);
    printf(_("Error deleting file %s.\n"
	     "Probably a few blocks were already missing from the database.\n"),
	   filename);
  }
  releaseClientSocket(sock);
  doneUtil();
  FREE(filename);
  if (ok == OK)
    return 0;
  else
    return -1;
}

/* end of gnunet-delete.c */
